### PR TITLE
Make quantiles configurable

### PR DESCRIPTION
**Pull request by wodity**

### CODE REVIEW

File: src/conn_handler.c [hunk 1]
```diff
@@ -136,9 +132,14 @@ static int stream_formatter(FILE *pipe, void *data, metric_type type, char *name
             STREAM("%s%s.upper|%f|%lld\n", prefix, name, timer_max(&t->tm));
             STREAM("%s%s.count|%lld|%lld\n", prefix, name, timer_count(&t->tm));
             STREAM("%s%s.stdev|%f|%lld\n", prefix, name, timer_stddev(&t->tm));
-            STREAM("%s%s.median|%f|%lld\n", prefix, name, timer_query(&t->tm, 0.5));
-            STREAM("%s%s.p95|%f|%lld\n", prefix, name, timer_query(&t->tm, 0.95));
-            STREAM("%s%s.p99|%f|%lld\n", prefix, name, timer_query(&t->tm, 0.99));
+            for (i=0; i < GLOBAL_CONFIG->num_quantiles; i++) {
+                if (GLOBAL_CONFIG->quantiles[i] == 0.5) {
+                    STREAM("%s%s.median|%f|%lld\n", prefix, name, timer_query(&t->tm, 0.5));
```
Comment (armon) [inline src/conn_handler.c]: We should probably `continue` after this to not emit both `median` and `p50`

File: src/conn_handler.c [hunk 2]
```diff
@@ -227,9 +228,11 @@ static int stream_formatter_bin(FILE *pipe, void *data, metric_type type, char *
             STREAM_BIN(BIN_TYPE_TIMER, BIN_OUT_STDDEV, timer_stddev(&t->tm));
             STREAM_BIN(BIN_TYPE_TIMER, BIN_OUT_MIN, timer_min(&t->tm));
             STREAM_BIN(BIN_TYPE_TIMER, BIN_OUT_MAX, timer_max(&t->tm));
-            STREAM_BIN(BIN_TYPE_TIMER, BIN_OUT_PCT | 50, timer_query(&t->tm, 0.5));
-            STREAM_BIN(BIN_TYPE_TIMER, BIN_OUT_PCT | 95, timer_query(&t->tm, 0.95));
-            STREAM_BIN(BIN_TYPE_TIMER, BIN_OUT_PCT | 99, timer_query(&t->tm, 0.99));
+            for (i=0; i < GLOBAL_CONFIG->num_quantiles; i++) {
+                STREAM_BIN(BIN_TYPE_TIMER, BIN_OUT_PCT |
+                    (int)(GLOBAL_CONFIG->quantiles[i] * 100),
```
Comment (armon) [inline src/conn_handler.c]: This may get to be a bit screwy given the lack of precision for things like 99.9%

**Comment by armon** [comment]
@wodity This is looking pretty good, but can you please add some tests for the config changes as well as an integration test for a custom quantile? Thanks.

File: src/conn_handler.c (hunk 1 above)
Comment (wodity) [inline src/conn_handler.c]: Adding p50 gives compatilbity with statsd, which is nice if you're moving from that to statsite.

**Comment by armon** [comment]
Thanks!

**Pull request merged**